internal/rest/avi_evh_obj.go

```diff
@@ -0,0 +1,805 @@
+
+				}
+			}
+		} else {
+			utils.AviLog.Debugf("key: %s, msg: sni child %s not found in cache, operation: POST", key, sni_node.Name)
+			_, rest_ops = rest.CACertCU(sni_node.CACertRefs, []avicache.NamespaceName{}, namespace, rest_ops, key)
+			_, rest_ops = rest.SSLKeyCertCU(sni_node.SSLKeyCertRefs, nil, namespace, rest_ops, key)
+			_, rest_ops = rest.PoolCU(sni_node.PoolRefs, nil, namespace, rest_ops, key)
+			_, rest_ops = rest.PoolGroupCU(sni_node.PoolGroupRefs, nil, namespace, rest_ops, key)
+			_, rest_ops = rest.HTTPPolicyCU(sni_node.HttpPolicyRefs, nil, namespace, rest_ops, key)
+
+			// Not found - it should be a POST call.
+			restOp := rest.AviVsBuildForEvh(sni_node, utils.RestPost, nil, key)
+			rest_ops = append(rest_ops, restOp...)
+		}
+		rest_ops = rest.SSLKeyCertDelete(sslkey_cert_delete, namespace, rest_ops, key)
+		rest_ops = rest.HTTPPolicyDelete(http_policies_to_delete, namespace, rest_ops, key)
+		rest_ops = rest.PoolGroupDelete(sni_pgs_to_delete, namespace, rest_ops, key)
+		rest_ops = rest.PoolDelete(sni_pools_to_delete, namespace, rest_ops, key)
+		utils.AviLog.Debugf("key: %s, msg: the SNI VSes to be deleted are: %s", key, cache_sni_nodes)
+	} else {
+		utils.AviLog.Debugf("key: %s, msg: sni child %s not found in cache and SNI parent also does not exist in cache", key, sni_node.Name)
+		_, rest_ops = rest.CACertCU(sni_node.CACertRefs, []avicache.NamespaceName{}, namespace, rest_ops, key)
+		_, rest_ops = rest.SSLKeyCertCU(sni_node.SSLKeyCertRefs, nil, namespace, rest_ops, key)
+		_, rest_ops = rest.PoolCU(sni_node.PoolRefs, nil, namespace, rest_ops, key)
+		_, rest_ops = rest.PoolGroupCU(sni_node.PoolGroupRefs, nil, namespace, rest_ops, key)
+		_, rest_ops = rest.HTTPPolicyCU(sni_node.HttpPolicyRefs, nil, namespace, rest_ops, key)
+
+		// Not found - it should be a POST call.
+		restOp := rest.AviVsBuildForEvh(sni_node, utils.RestPost, nil, key)
+		rest_ops = append(rest_ops, restOp...)
+	}
+	return cache_sni_nodes, rest_ops
+}
+
+func (rest *RestOperations) AviVsBuildForEvh(vs_meta *nodes.AviEvhVsNode, rest_method utils.RestMethod, cache_obj *avicache.AviVsCache, key string) []*utils.RestOp {
+	if !vs_meta.EVHParent {
+		rest_ops := rest.AviVsChildEvhBuild(vs_meta, rest_method, cache_obj, key)
+		return rest_ops
+	} else {
+		// This is EVH Parent
+		network_prof := "/api/networkprofile/?name=" + vs_meta.NetworkProfile
+		app_prof := "/api/applicationprofile/?name=" + vs_meta.ApplicationProfile
+		name := vs_meta.Name
+		cksum := vs_meta.CloudConfigCksum
+		checksumstr := strconv.Itoa(int(cksum))
+		cr := lib.AKOUser
+		cloudRef := "/api/cloud?name=" + utils.CloudName
+		svc_mdata_json, _ := json.Marshal(&vs_meta.ServiceMetadata)
+		svc_mdata := string(svc_mdata_json)
+		vrfContextRef := "/api/vrfcontext?name=" + vs_meta.VrfContext
+		seGroupRef := "/api/serviceenginegroup?name=" + lib.GetSEGName()
+		enableRHI := lib.GetEnableRHI() // We don't impact the checksum of the VS since it's a global setting in AKO.
```

Comment:
@chauhanshubham you have changed this logic right? Where you are populating the RHI value in layer 2?

Reply:
enableRhi, seGroupRef both need updates

Reply:
Done. Thanks!